### PR TITLE
Move sysconf and confstr to native code

### PR DESCRIPTION
**Pull request by sbc100**
As part of this change `thread::hardware_concurrency` now returns the
correct value under node by calling `emscripten_num_logical_cores`.
Previously `_SC_NPROCESSORS_ONLN` had a partial reimplemntation of
`emscripten_num_logical_cores`.

This means that `_SC_NPROCESSORS_ONLN` now returns 1 when `USE_PTHREADS`
is not used (since that is what `emscripten_num_logical_cores` does).

Aside form that I've tried to keep the actual return values of confstr
and sysconf relatively constant.

### CODE REVIEW

File: system/lib/standalone/standalone.c [hunk 1]
```diff
@@ -117,15 +117,22 @@ void *emscripten_memcpy_big(void *restrict dest, const void *restrict src, size_
 
 extern void emscripten_notify_memory_growth(size_t memory_index);
 
+size_t emscripten_get_heap_size() {
+  return __builtin_wasm_memory_size(0) * WASM_PAGE_SIZE;
+}
+
+size_t emscripten_get_heap_max() {
+  return emscripten_get_heap_size();
```
Comment (kripken) [inline system/lib/standalone/standalone.c:125]: Maybe add a comment that we can't do better?

File: tests/unistd/pathconf.out [hunk 2]
```diff
@@ -49,13 +49,13 @@ errno: 0
...
-_PC_REC_INCR_XFER_SIZE: -1
+_PC_REC_INCR_XFER_SIZE: 4096
 errno: 0
 
-_PC_REC_MAX_XFER_SIZE: -1
+_PC_REC_MAX_XFER_SIZE: 4096
 errno: 0
 
-_PC_SYMLINK_MAX: -1
+_PC_SYMLINK_MAX: 255
```
Comment (kripken) [inline tests/unistd/pathconf.out:58]: is it not worth matching the old values?

Comment (sbc100) [reply tests/unistd/pathconf.out:58]: I kind of decided for some values is wasn't worth it, better to be in line with musl... but maybe its better to keep this change clean.. I'll do that.

Comment (kripken) [reply tests/unistd/pathconf.out:58]: I don't feel strongly either way. But it does feel safer to change less.

(If you prefer to change some values, that's fine too, but let's document that in the changelog then.)

**Comment by sbc100** [comment]
Thanks, I made this into a no-op and we can look at modify some of the conf values as a follow.

**Review by kripken — APPROVED** [review]

**Pull request merged**